File: frontend/src/types/note.ts
export interface AudioNote {
  id: string
  title: string
  tags: string[]
  notes?: string
  audio_filename: string
  audio_path?: string
  audio_duration?: number
  transcription?: string
  summary?: string
  status: 'pending' | 'processing' | 'completed' | 'error'
  created_at: Date
  updated_at: string
}

export interface CreateNoteRequest {
  title: string
  tags: string[]
  notes?: string
}

export interface UpdateNoteRequest {
  title?: string
  tags?: string[]
  notes?: string
}

export interface NoteResponse {
  id: string
  title: string
  tags: string[]
  notes?: string
  audio_filename: string
  audio_url?: string
  status: string
  transcription?: string
  summary?: string
  created_at: string
  updated_at: string
}

File: backend/schemas/note.py
from pydantic import BaseModel, UUID4
from typing import List, Optional
from datetime import datetime


class NoteBase(BaseModel):
    title: str
    tags: List[str] = []
    notes: Optional[str] = None


class NoteCreate(NoteBase):
    pass


class NoteUpdate(NoteBase):
    title: Optional[str] = None
    tags: Optional[List[str]] = None
    notes: Optional[str] = None

    class Config:
        extra = "ignore"


class NoteResponse(NoteBase):
    id: UUID4
    audio_filename: str
    audio_path: Optional[str]
    transcription: Optional[str]
    summary: Optional[str]
    status: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True